tfhe/src/shortint/engine/server_side/scalar_sub.rs

```diff
@@ -37,6 +37,9 @@ impl ShortintEngine {
         ct: &mut Ciphertext,
         scalar: u8,
     ) -> EngineResult<Ciphertext> {
+        if !server_key.is_scalar_sub_possible(ct, scalar) {
+            self.message_extract_assign(server_key, ct)?;
+        }
         let mut ct_result = ct.clone();
         self.smart_scalar_sub_assign(server_key, &mut ct_result, scalar)?;
```

Comment:
unchecked call ?

Reply:
If we always have at least one bit of carry, I guess this is safe.
Is it always the case?

Reply:
nope good point